apisix/discovery/kubernetes/informer_factory.lua

```diff
@@ -152,48 +152,47 @@ local function split_event (body, dispatch_event)
         remainder_body = string.sub(body, captured_size + 1)
     end
 
-    return remainder_body, "Success", nil
+    return remainder_body, "Success"
```

Comment:
```lua   
         remainder_body, reason, err = split_event(body, dispatch_event, informer)
            if reason ~= "Success" then
                if reason == "ResourceGone" then
                    return true
                end
                return false, reason, err
            end
```

"Success" makes sense here because the code above will use it.
If there is no "Success", 
how can only use remain_body is nil as a detection method, 
I think this is not a good code style

Reply:
In Lua land, we mainly use this way: `local ok, body, err = ... ...`

Reply:
Ok，I will  update

Reply:
@membphis 
Let's Imagine,  if we use  local ok ,body ,err  way :

case error  -> return  false , remainer_body, reason, err
case ok -> return true, remainer_body 

and  code in  watch function will like 
```lua
ok, remainer_body, reason, err = split_event(body, dispatch_event, informer)
if not ok  then
    if reason == "ResourceGone" then
        return true
    end
    return false, reason, err
end

.....
```

is this what you expect